backend/src/models/upload.py

```diff
@@ -1,18 +1,40 @@
-from sqlalchemy.orm import mapped_column, Mapped, relationship
-from sqlalchemy import ForeignKey
+from enum import Enum
 
+from sqlalchemy import ForeignKey
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from src.models import BaseModel
 
 
+class CeleryTaskStatus(str, Enum):
+    #: Task state is unknown (assumed pending since you know the id).
+    PENDING = "PENDING"
+    #: Task was received by a worker (only used in events).
+    RECEIVED = "RECEIVED"
+    #: Task was started by a worker (:setting:`task_track_started`).
+    STARTED = "STARTED"
+    #: Task succeeded
+    SUCCESS = "SUCCESS"
+    #: Task failed
+    FAILURE = "FAILURE"
+    #: Task was revoked.
+    REVOKED = "REVOKED"
+    #: Task was rejected (only used in events).
+    REJECTED = "REJECTED"
+    #: Task is waiting for retry.
+    RETRY = "RETRY"
+    IGNORED = "IGNORED"
+
+
 class Upload(BaseModel):
```

Comment:
I'm not sure about this. When would this field be updated? If we want this Status we can just ask celery. I would suggest having just 
- error on exception
- ready (when upload is ready)
- pending when upload is being processed.

Why
1) We need **upload** status not **task** status for frontend
2) There is no point in keeping this alongside redis
3) I'm not sure in what cases this would be updated
4) If this would be left, we would need to somehowe decide how to map 9 statuses into one and this would need to be done somewhere

Reply:
The task status will be stored in the `upload` table.

Reply:
We wont have to, they will be updated during processing. And I don't agree on leaving this like this since its not clear when and how this state will be updated and why would we keep the same state in both databases

Reply:
Ok, I see your point. But what about users' history? The status will be deleted from Redis after some time.

Reply:
What's your suggestion then?

Reply:
Maybe we can store the task statuses in Redis for 30 days and users' history for the same amount of time?

Reply:
Like i said,
User uploads a file
-> set status to pending in postgres
Exception during processing is raised
-> set status to error in postgres
Frontend asks for status and celery status is SUCCESS 
-> set status as done  in postgres

History will return fields from postgres and uploads with pending status will ask for updates celery